File: docs/extensibility/codesnippet/VisualBasic/walkthrough-implementing-code-snippets_9.vb
        Friend Sub New(ByVal textViewAdapter As IVsTextView, ByVal textView As ITextView, ByVal provider As TestCompletionHandlerProvider)
            Me.m_textView = textView
            Me.m_provider = provider
            Me.m_vsTextView = textViewAdapter

            Dim textManager As IVsTextManager2 = DirectCast(m_provider.ServiceProvider.GetService(GetType(SVsTextManager)), IVsTextManager2)
            textManager.GetExpansionManager(m_exManager)
            m_exSession = Nothing

            'add the command to the command chain
            textViewAdapter.AddCommandFilter(Me, m_nextCommandHandler)
        End Sub

File: docs/extensibility/codesnippet/CSharp/walkthrough-implementing-code-snippets_9.cs
        internal TestCompletionCommandHandler(IVsTextView textViewAdapter, ITextView textView, TestCompletionHandlerProvider provider)
        {
            this.m_textView = textView;
            m_vsTextView = textViewAdapter;
            m_provider = provider;
            //get the text manager from the service provider
            IVsTextManager2 textManager = (IVsTextManager2)m_provider.ServiceProvider.GetService(typeof(SVsTextManager));
            textManager.GetExpansionManager(out m_exManager);
            m_exSession = null;

            //add the command to the command chain
            textViewAdapter.AddCommandFilter(this, out m_nextCommandHandler);
        }